### PR TITLE
Add a PyPI upload action

### PR DESCRIPTION
**Pull request by kklein**
# Checklist

- [ ] Added a `CHANGELOG.rst` entry

### CODE REVIEW

**Review by FrancescMartiEscofetQC — DISMISSED** [review]

File: .github/workflows/package.yml [hunk 1]
```diff
@@ -12,7 +12,27 @@ jobs:
...
+    if: github.event_name == 'release' && github.event.action == 'published'
+    steps:
+      - uses: actions/download-artifact@v4
+        with:
+          name: artifact
+          path: dist
+      - uses: pypa/gh-action-pypi-publish@v1.9.0
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_TOKEN }}
```
Comment (pavelzw) [inline .github/workflows/package.yml:38]:
```suggestion
```

you can use OIDC https://github.com/Quantco/multiregex/pull/52#issuecomment-2127001861

File: pixi.toml [hunk 2]
```diff
@@ -12,8 +12,6 @@ pre-commit = "*"
 git_root = "*"
 pip = "*"
 python = ">=3.10"
-setuptools-scm = "*"
-setuptools = ">=61"
```
Comment (pavelzw) [inline pixi.toml:16]: these are actually still needed for the `postinstall` task

Comment (kklein) [reply pixi.toml:16]: Ah, I see!

**Review by pavelzw — APPROVED** [review]
As long as OIDC is set up properly, this looks good 👍🏻

**Pull request merged**